docs/linting/troubleshooting/Formatting.mdx

```diff
@@ -59,4 +52,21 @@ module.exports = {
 };
 ```
 
-Note that even if you don't use `prettier`, you can use `eslint-config-prettier` as it exclusively turns **off** all formatting rules.
+Note that even if you use a formatter other than `prettier`, you can use `eslint-config-prettier` as it exclusively turns **off** all formatting rules.
+
+## ESLint Core and Formatting
+
+Most lint rules fall into one of two to three categories:
+
+- **Logical**: Rules that care about the logic in runtime behavior of code.
+- **Stylistic**: Rules that care about style concerns which do impact runtime behavior of code, but generally not logic. These are mostly around naming or which roughly-equivalent syntax constructs to use (such as function declarations vs. arrow functions).
+  - **Formatting**: Stylistic rules that care only about trivia (semicolons, whitespace, etc.) without impacting the runtime behavior of the code. These rules conflict with dedicated formatters such as Prettier.
```

Comment:
@rubiesonthesky replying to https://github.com/typescript-eslint/typescript-eslint/pull/6510#issuecomment-1438810598 here because it's a good one to bring up and I want to thread:

> I feel weird mentioning semicolons as just formatting, because they can affect runtime behavior. 😅 The decision between using them only where needed and always, is stylistic. Of course this is nit and it may be clear for everyone what is meant here.

You're correct that semicolons can impact runtime behavior - but it's only in rare cases that are almost never hit anymore (`return\n{`). I think most folks reading this would understand that the intent of the rules is to enforce style of using whitespace, spaces/tabs, etc.

Aside: technically, whitespace can also impact runtime too! `(function () {}).toString().indexOf("(")`.toString() 😄.